src/control-plane/backend/lambda/listen-state-status/index.ts

```
@@ -0,0 +1,300 @@
+};
+
+const getPipeline = async (pipelineId: string) => {
+  try {
+    const type = `PIPELINE#${pipelineId}#latest`;
+    const input: QueryCommandInput = {
+      TableName: clickStreamTableName,
+      IndexName: prefixTimeGSIName,
+      KeyConditionExpression: '#prefix= :prefix',
+      FilterExpression: 'deleted = :d AND #type = :type',
+      ExpressionAttributeNames: {
+        '#prefix': 'prefix',
+        '#type': 'type',
+      },
+      ExpressionAttributeValues: {
+        ':d': false,
+        ':prefix': 'PIPELINE',
+        ':type': type,
+      },
+    };
+    const records: Record<string, NativeAttributeValue>[] = [];
+    for await (const page of paginateQuery({ client: docClient }, input)) {
+      records.push(...page.Items as Record<string, NativeAttributeValue>[]);
+    }
+    if (records.length > 0) {
+      return records[0];
+    }
+    return;
+  } catch (err) {
+    logger.error('Failed to query pipeline: ', { pipelineId, err });
+    return;
+  }
+};
+
+const updatePipelineStateStatus = async (
+  projectId: string, pipelineId:string,
+  eventDetail: StepFunctionsExecutionStatusChangeNotificationEventDetail) => {
+  try {
+    const executionDetail: ExecutionDetail = {
+      executionArn: eventDetail.executionArn,
+      name: eventDetail.name,
+      status: eventDetail.status,
+    };
+    const input: UpdateCommandInput = {
+      TableName: clickStreamTableName,
+      Key: {
+        id: projectId,
+        type: `PIPELINE#${pipelineId}#latest`,
+      },
+      UpdateExpression: 'SET #executionDetail = :executionDetail',
+      ExpressionAttributeNames: {
+        '#executionDetail': 'executionDetail',
+      },
+      ExpressionAttributeValues: {
+        ':executionDetail': executionDetail,
+      },
+    };
+    await docClient.send(new UpdateCommand(input));
+  } catch (err) {
+    logger.error('Failed to update pipeline state status: ', { projectId, pipelineId, err });
```

This swallows this event. I prefer to throw the exception to retry it later.